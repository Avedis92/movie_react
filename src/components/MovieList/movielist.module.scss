@import "../sasscommonmixxins/flex.scss";
@import "../sasscommonmixxins/button.scss";
@import "../sasscommonmixxins/mediaqueries.scss";

.movieListContainer {
  background-color: rgb(224, 224, 224);
  .movieGridsContainer {
    margin-left: 5%;
    margin-right: 5%;
    display: grid;
    gap: 30px;
    @include mobile {
      @include flexElements(flex, column, center, space-evenly);
      flex-wrap: wrap;
    }
  }
  .seeMore {
    margin-top: 5%;
    margin-left: 45%;
    margin-bottom: 5%;
    cursor: pointer;
    @include styleButtons(10%, 50px, white, rgb(0, 195, 255), 10px, none);
    transition: 500ms;
    &:hover {
      transform: scale(110%);
    }
    @include mobile {
      width: 40vw;
      margin-left: 30vw;
    }
  }
}
