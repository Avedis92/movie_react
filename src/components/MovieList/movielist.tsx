import MovieCard from '../MovieCard/moviecard';
import HorizontalMovieCard from '../HorizontalMovieCard/horizontalmoviecard';
import ErrorHandler from '../ErrorHandler/errorhandler';
import Loader from '../Loader/loader';
import styles from './movielist.module.scss';
import * as constants from '../../constants/constants';

interface MovieLists {
  id:number,
  genre_ids:Array<number>,
  original_title:string,
  vote_average:number,
  overview:string,
  poster_path:string
}
function MovieList({
  movieList, gridStyle, gridTemplateColumn, errorMessage, setMoviePage, listLoading, loadingMore,
}:{movieList:Array<MovieLists>, gridStyle:string, listLoading:boolean, loadingMore:boolean, gridTemplateColumn:number, errorMessage:string, setMoviePage:()=>void}) {
  const {
    movieListContainer, movieGridsContainer, seeMore,
  } = styles;

  return (
    <div className={movieListContainer}>
      {!listLoading ? (
        <div className={movieGridsContainer} style={{ gridTemplateColumns: `repeat(${gridTemplateColumn},1fr)` }}>
          {movieList.length !== 0 && gridStyle === 'Multi' ? movieList.map((myMovies, index) => (
            <MovieCard
              key={index}
              id={myMovies.id}
              genreIds={myMovies.genre_ids}
              title={myMovies.original_title}
              voteAverage={myMovies.vote_average}
              overview={myMovies.overview}
              posterPath={constants.postPathUrl + myMovies.poster_path}
            />
          )) : movieList.length !== 0 && gridStyle === 'Horizontal' ? movieList.map((myMovies, index) => (
            <HorizontalMovieCard
              key={index}
              id={myMovies.id}
              genreIds={myMovies.genre_ids}
              title={myMovies.original_title}
              voteAverage={myMovies.vote_average}
              overview={myMovies.overview}
              posterPath={constants.postPathUrl + myMovies.poster_path}
            />
          )) : <ErrorHandler message={errorMessage} />}
        </div>
      ) : <Loader />}
      {
        loadingMore ? (
          <Loader />)
          : <button className={seeMore} type="button" onClick={() => setMoviePage()}>See More</button>
        }
    </div>
  );
}
export default MovieList;
