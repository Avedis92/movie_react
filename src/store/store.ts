import { configureStore } from '@reduxjs/toolkit';
import genreReducer from './slices/genre';

export const store = configureStore({
  reducer: {
    genre: genreReducer,
  },
});

export type RootState=ReturnType<typeof store.getState>;
export type MovieDispatch=typeof store.dispatch;
