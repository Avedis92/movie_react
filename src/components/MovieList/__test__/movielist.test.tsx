import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../../store/store';
import MovieList from '../movielist';

const movie = {
  id: 297761,
  genre_ids: [14, 28, 80],
  original_title: 'Suicide Squad',
  vote_average: 5.91,
  overview: 'From DC Comics comes the Suicide Squad, an antihero team of incarcerated supervillains who act as deniable assets for the United States government, undertaking high-risk black ops missions in exchange for commuted prison sentences.',
  poster_path: '/e1mjopzAS2KNsvpbpahQ1a6SkSn.jpg',
};
test('Make a snapshot test', () => {
  const {asFragment} = render(
    <Provider store={store}>
      <MovieList
        movieList={[movie]}
        gridStyle="Multi"
        gridTemplateColumn={4}
        errorMessage="Error detected"
        setMoviePage={() => null}
        listLoading={false}
        loadingMore={false}
      />
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
