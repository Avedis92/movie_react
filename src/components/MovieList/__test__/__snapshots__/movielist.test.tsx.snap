// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="movieListContainer"
  >
    <div
      class="movieGridsContainer"
      style="grid-template-columns: repeat(4,1fr);"
    >
      <div
        class="container"
      >
        <div
          class="movieItem"
        >
          <img
            alt="Suicide Squad"
            src="https://image.tmdb.org/t/p/w500/e1mjopzAS2KNsvpbpahQ1a6SkSn.jpg"
          />
          <div
            class="movieInfo"
          >
            <div
              class="titleAndGenres"
            >
              <h3
                data-testid="movieTitle"
              >
                Suicide Squad
              </h3>
            </div>
            <div
              class="votesAverage"
            >
              <p>
                5.91
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      class="seeMore"
      type="button"
    >
      See More
    </button>
  </div>
</DocumentFragment>
`;
